src/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs

```diff
@@ -18,7 +18,11 @@ internal static class UnixCommandLinePing
 
         private static readonly string s_discoveredPing4UtilityPath = GetPingUtilityPath(ipv4: true);
         private static readonly string s_discoveredPing6UtilityPath = GetPingUtilityPath(ipv4: false);
-        private static readonly bool s_isOSX = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+        private static readonly bool s_isBSD = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD"));
+        private static Lazy<bool> s_isBusybox = new Lazy<bool>(() => IsBusyboxPing(s_discoveredPing4UtilityPath));
+
+        [DllImport(Interop.Libraries.SystemNative, EntryPoint = "SystemNative_ReadLink", SetLastError = true)]
+        private static extern unsafe int ReadLink(string path, byte* buffer, int bufferSize);
```

Comment:
We should instead just include https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/Interop/Unix/System.Native/Interop.ReadLink.cs in whatever projects include this file, and use that ReadLink method.  That then also addresses my other comments about the 4000, the stackalloc, etc., as that's all handled in that file.

Reply:
I started with that but I had difficulties to resolve all dependencies for ArrayPool as that depends on TLS and others. It look strange to pull in so much just to make one simple call.  I can certainly cut the size down. The main goal is to detect normal use of busybox and that should fit easily to 100 bytes. 
Other alternative I considered was actually running ping utility and check if the option is supported.

Reply:
PATH_MAX is 4096 in linux/limits.h. However I figured out how to resolve dependencies for ReadLink by adding reference instead of including files as I was trying to before.